boards/DCM/Test/Src/Test_StateMachine.cpp

```diff
@@ -399,4 +418,157 @@ TEST_F(
     }
 }
 
+// DCM-8
+TEST_F(
+    DcmStateMachineTest,
+    regen_allowed_only_when_going_faster_than_5kph_and_airs_closed)
+{
+    SetInitialState(App_GetDriveState());
+
+    // Turn the DIM start switch on to prevent state transitions in
+    // the drive state.
+    App_CanRx_DIM_SWITCHES_SetSignal_START_SWITCH(
+        can_rx_interface, CANMSGS_DIM_SWITCHES_START_SWITCH_ON_CHOICE);
+
+    // Press down accelerator pedal and regen paddle
+    App_CanRx_FSM_PEDAL_POSITION_SetSignal_MAPPED_PEDAL_PERCENTAGE(
+        can_rx_interface, 60.0f);
+    App_CanRx_DIM_REGEN_PADDLE_SetSignal_MAPPED_PADDLE_POSITION(
+        can_rx_interface, 50.0f);
+
+    float expected_torque_request_value =
+        60.0f / 100.0f * MAX_SAFE_TORQUE_REQUEST_NM;
+    float expected_regen_request_value =
+        -50.0f / 100.0f * MAX_SAFE_TORQUE_REQUEST_NM;
+
+    // Close AIRs and assert regen is still off
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_CLOSED_CHOICE);
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE);
+    LetTimePass(state_machine, 10);
+
+    AssertFloatsEqual(
+        expected_torque_request_value,
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface));
+
+    // Check regen doesn't turn on when only one wheel is going faster than 5kph
+    float threshold_wheel_speed = 5.0f;
+    float value_over_threshold_wheel_speed =
+        std::nextafter(5.0f, std::numeric_limits<float>::max());
+
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_LEFT_WHEEL_SPEED(
+        can_rx_interface, threshold_wheel_speed);
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_RIGHT_WHEEL_SPEED(
+        can_rx_interface, value_over_threshold_wheel_speed);
+
```

Comment:
```suggestion
```

Reply:
easier to read this way. Kinda conveys to the reader that the signals are all related to a particular test condition